deploy/vagrant/core/EnvSetup.rb

```diff
@@ -57,3 +57,45 @@ def config_hosts(name)
     hosts.close unless hosts == nil
   end
 end
+
+require 'yaml'
+require 'open-uri'
+
+def get_tachyon_home
+  @v = YAML.load_file('tachyon_version.yml')
+  
+  type = @v['Type']
+  if type == "Local"
+    puts 'using local tachyon dir'
+    return "../../"
+  end
+
+  home = "../tachyon"
+  if type == "Github"
+    repo = @v['Github']['Repo']
+    hash = @v['Github']['Hash']
+    url = "#{repo}/commit/#{hash}"
+
+    begin
+      open(url)
+    rescue OpenURI::HTTPError
+      $stderr.print "The github URL #{url} is invalid"
+      exit(1)
+    end
+
+    puts "git clone #{url} and mvn package"
+    `if [ -d ../tachyon ]; then rm -rf ../tachyon; fi`
+    `mkdir -p ../tachyon`
+    `pushd ../tachyon > /dev/null; \
+     git init; \
+     git remote add origin #{repo}; \
+     git fetch origin; \
+     git checkout #{hash}; \
+     mvn package -DskipTests; \
```

Comment:
The problem with the first solution is there usually will also be a target hdfs version. For the second suggestion, isn't it the same as just having the user use local instead of github? (ie. checkout the commit they want, build it, and then run the deployment script). 

Since that is an option, we probably don't need to add anything here.

If most people are using higher versions of hadoop, we can change it later to be more user friendly.